pkg/forwarder/forwarder_health.go

```diff
@@ -111,17 +110,18 @@ func (fh *forwarderHealth) healthCheckLoop() {
 	}
 }
 
-// computeDomainsURL populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
-func (fh *forwarderHealth) computeDomainsURL() {
-	for domain, apiKeys := range fh.keysPerDomains {
+// computeKeysPerAPIDomain populates a map containing API Endpoints per API keys that belongs to the forwarderHealth struct
+func (fh *forwarderHealth) computeKeysPerAPIDomain(keysPerDomains map[string][]string) {
+	fh.keysPerAPIDomain = make(map[string][]string)
+	re := regexp.MustCompile("datadoghq.[a-z]*")
```

Comment:
now that I think a bit more about it, such a wide regex may not be a great idea, we may end up matching more than we'd want.  Typically, for a domain of `https://datadoghq.customdomain.com`, I'm not sure whether we should change the API domain, so to stay safe I think we should leave it unchanged for now. Same uncertainty if the domain is set to `https://us1.datadoghq.com` for example. We can revisit in a future agent version if needed.

Since we maintain a hardcoded list of known `ddURLs` in the config package (https://github.com/DataDog/datadog-agent/blob/8890126df2160940d9a754a57f52a756162ae957/pkg/config/config.go#L623-L628), I think we should match against those only instead, with a similar logic as the `config.AddAgentVersionToDomain` function.